.github/workflows/CI.yml

```diff
@@ -0,0 +1,65 @@
+name: CI
+
+on:
+  push:
+    branches: [ "main" ]
+    paths:
+      - project/**
+  pull_request:
+    branches: [ "main" ]
+    paths:
+      - project/**
+
+  # Allows to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets the GITHUB_TOKEN permissions to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: 'pages'
+  cancel-in-progress: true
+
+jobs:
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    defaults:
+        run:
+          working-directory: project
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Node
+        uses: actions/setup-node@v4
+        with:
+          cache: 'npm'
+          cache-dependency-path: project/package-lock.json
```

Comment:
I'm not sure either, but I partly based it on one of our previous projects (https://github.com/RustyRipper/online-city-tickets/blob/main/.github/workflows/frontend.yml), where it was also like this and worked. It may not be required, but it shouldn't get in the way.